Refactorings for factory, urls and unit tests

dfoderick opened the pull request:
Large diff but most of it is refactoring testdata into separate files

dfoderick committed:
refactor factory

dfoderick committed:
simplify the api factory

dfoderick committed:
move test skip logic into testdata

dfoderick committed:
refactor factory, tests and urls

brentongunning committed:
Rename outputNumber to outputIndex

dfoderick committed:
test coverage

dfoderick committed:
refactor test skip logic

dfoderick committed:
merged outputnumber => outputindex

brentongunning commented on src/index.js:
```diff
@@ -27,6 +27,5 @@ export {
   BsvInsightApi,
-  BSV_MAINNET_URL,
-  BSV_TESTNET_URL
+  BSV_BCHSVEXPLORER_MAINNET_URL
```
Let's remove all but Insight. If we need to add any more back later, we can do that at the time, but this will be super nice to just have one global export

brentongunning commented on test/integration/api.integration.test.js:
```diff
@@ -1,275 +1,329 @@
-          expect(res).toBeGreaterThan(0)
-        })
+// eslint-disable-next-line no-restricted-syntax
+for (const testdata of data) {
```
Sorry Dave. Can we change this back to forEach? Didn't realize lint would complain

ClemensLey commented:
LGTM

brentongunning commented:
LGTM

dfoderick committed:
refactored based on code review feedback

dfoderick committed:
only export insight

dfoderick commented:
I made all changes from code review feedback
* only export Insight
* removed Urls file
* moved tests for test-only function to meta_integration.tests.js
* improved checkskip function to remove duplicate
* made constructor with 3 parameters (2 optional) instead of apiconfig object

ClemensLey commented:
LGrTM
